pkg/aws/services/route53/route53.go

```diff
@@ -0,0 +1,68 @@
+package route53
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/giantswarm/microerror"
+)
+
+type CNAME struct {
+	Name  string
+	Value string
+}
+
+func (s *Service) FindHostedZone(basename string) (string, error) {
+	s.scope.Info("Searching route53 hosted zone ID")
+
+	output, err := s.Client.ListHostedZonesByName(&route53.ListHostedZonesByNameInput{
```

Comment:
just a note, I'm not sure if we need additional IAM permission to list route53domains.
I don't think we have it by default https://github.com/giantswarm/giantswarm-aws-account-prerequisites/blob/master/aws-operator-role/iam-policy.json but I might be wrong

Reply:
The additional permissions are already listed in customer documentation

Reply:
Ok than I think https://github.com/giantswarm/giantswarm-aws-account-prerequisites/blob/master/aws-operator-role/iam-policy.json needs to updated. I don't know if this is still used when creating a new installation

Reply:
ok I might be missing something. Isn't this a dedicated IAM role that the customer handles?
Like we describe here https://docs.giantswarm.io/advanced/iam-roles-for-service-accounts/#additional-iam-role-permissions-on-giantswarmawsoperator-needed

Reply:
That is correct, but we do mention this link I pasted in our official docs here https://docs.giantswarm.io/getting-started/cloud-provider-accounts/aws/#iam-aws-operator-role-permissions.

When customer needs to create GiantSwarmAWSOperator for the first time and here we're missing the IAM permissions.